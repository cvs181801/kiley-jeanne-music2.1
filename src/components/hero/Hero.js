import React, {useState, useEffect} from 'react'
import { client } from '../../client';
import './hero.css'

export default function Hero() {
    const [isHeroLoading, setIsHeroLoading] = useState(true); 
    const [heroContent, setHeroContent] = useState({});

    const getHeroContent = async ()=> {
        try{
            const response = await client.getEntries({content_type:'hero'})
   
            const title = response.items[0].fields.title;
            const image = response.items[0].fields.image.fields.file.url;  //  var imageURL = 'https:' + asset.fields.file.url + '?w=200&h=200';
            const alt = response.items[0].fields.image.fields.description;
            const tagline = response.items[0].fields.tagline.content[0].content[0].value;
            setIsHeroLoading(false)
            setHeroContent({
                title: title,
                image: image,
                alt: alt,
                tagline: tagline
            })
    
  
        }
        catch(error){
            console.log(error) 
            setIsHeroLoading(false)
            setHeroContent({
                title: 'Sorry! Something went wrong.  Please try again.',
                image: '',
                alt: 'sorry, something went wrong',
                tagline: 'darn!'
            })
        }
    }

    useEffect(()=>{
        getHeroContent()
    },[])


  return (
    
    <div
        className="containerDivHero"
    >
        <section
            id="heroSection"
        >
        
            <div
                className="topDiv"
            >
                {/* <p
                    className="welcomeDiv"
                >Welcome â™¡
                </p> */}

                    <h1>{heroContent.title}</h1>
                
            </div>
                    <div
                        className="imgDiv"
                            style={{
                                backgroundImage: `url(${heroContent.image})`,
                            }}
                        >
                            <p
                                className="imgDiv-p"
                            >{isHeroLoading ? 'Loading...' : heroContent.tagline}</p>
                    </div>
            
            <div
                className="bottomDiv"
            >
            </div>
        </section>
    </div>
  )
}

