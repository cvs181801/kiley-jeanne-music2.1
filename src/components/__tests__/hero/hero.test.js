import {render, screen, cleanup} from '@testing-library/react'
import '@testing-library/jest-dom'
import Hero from '../../hero/Hero'
import renderer from 'react-test-renderer'

afterEach(()=> {
    cleanup();
})

test('should render hero tagline', ()=> {
    render(<Hero/>)
    const heroTagline = screen.getByText(/Loading.../i)
    expect(heroTagline).toBeInTheDocument();
})

test('should render the hero component', ()=> {
    const component = renderer.create(
        <Hero/>,
      );
      let tree = component.toJSON();
      expect(tree).toMatchSnapshot();
})


//write a test to test the API