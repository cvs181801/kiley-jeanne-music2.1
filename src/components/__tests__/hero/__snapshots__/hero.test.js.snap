// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the hero component 1`] = `
<div
  className="containerDivHero"
>
  <section
    id="heroSection"
  >
    <div
      className="topDiv"
    >
      <h1 />
    </div>
    <div
      className="imgDiv"
      style={
        Object {
          "backgroundImage": "url(undefined)",
        }
      }
    >
      <p
        className="imgDiv-p"
      >
        Loading...
      </p>
    </div>
    <div
      className="bottomDiv"
    />
  </section>
</div>
`;
